backend/apps/notifications/views.py

```
@@ -1,24 +1,51 @@
+    n.save(update_fields=["read"])
+    return serialize_inapp_notification(n)
+
+
+@router.post("/inapp/mark-all-read/", response={200: dict})
+def mark_all_inapp_read(request):
+    updated = InAppNotification.objects.filter(user=request.user, read=False).update(read=True)
+    return 200, {"marked": updated}
+
+
+@router.get("/", response=list[EmailNotificationSchema])
 def list_email_notifications(request):
```

Ruff (0.15.0)

[warning] Unused function argument: `request`

(ARG001)